src/Components/test/E2ETest/ServerExecutionTests/InteropReliabilityTests.cs

```diff
@@ -581,21 +510,17 @@ public async Task DispatchingEventsWithInvalidEventHandlerId()
         public async Task EventHandlerThrowsSyncExceptionTerminatesTheCircuit()
         {
             // Arrange
-            await GoToTestComponent(Batches);
-            var sink = _serverFixture.Host.Services.GetRequiredService<TestSink>();
-            var logEvents = new List<(LogLevel logLevel, string eventIdName, Exception exception)>();
-            sink.MessageLogged += (wc) => logEvents.Add((wc.LogLevel, wc.EventId.Name, wc.Exception));
 
             // Act
             await Client.ClickAsync("event-handler-throw-sync", expectRenderBatch: false);
 
             await Task.Delay(1000);
```

Comment:
Could we replace this with something like `Client.ExpectLog("CircuitUnhandledException", () => ..., LogMessages)` where we essentially poll every 500ms for DefaultOperationTimeout so that we can avoid the flakyness?

Reply:
We can just wrap this in `ExpectCircuitError`, which is basically the same thing.